libft/tests/map/mapss/ft_mapss_print_test.c

```diff
@@ -6,11 +6,87 @@
 /*   By: inikulin <inikulin@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/08/13 21:34:36 by inikulin          #+#    #+#             */
-/*   Updated: 2024/08/13 21:49:35 by inikulin         ###   ########.fr       */
+/*   Updated: 2024/10/27 00:39:00 by inikulin         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
-void	ft_mapss_print_test(void)
+#include "../../tests_internal.h"
+
+#define START 0
+#define SZ 1
+//#define DEBUG
+
+static void	catch(char* fname, int *out, int *save)
+{
+	remove(fname);
+	fflush(stdout);
+	*out = open(fname, O_RDWR|O_CREAT, 0600);
+	assert (-1 != *out);
+	*save = dup(fileno(stdout));
+	assert (-1 != dup2(*out, fileno(stdout)));
+}
+
+static void	finally(int *out, int *save)
 {
-	return ;
+	fflush(stdout);
+    close(*out);
+	dup2(*save, fileno(stdout));
+	close(*out);
+}
+
+typedef struct s_testcase
+{
+	char	***content;
+	int		sz;
+	char	*exp_res;
+}	t_testcase;
+
+int	ft_mapss_print_test(void)
+{
+	int out, save;
+	remove("custom.tmp");
+	t_testcase tests[SZ];
+	char **a = ft_s2("a", "aaa");
+	char **b = ft_s2("b", "");
+	char **c = ft_s2("", "ccc");
+	char ***t1 = ft_ss3(a, b, c);
+	tests[0] = (t_testcase){t1, 3, "Map size: [3]\nkey: [], value: [ccc]\nkey: [a], value: [aaa]\nkey: [b], value: []\n"};
+
+	#ifdef DEBUG
+	printf("Starting %d tests.\n", SZ);
+	#endif
+	for (int i = START; i < SZ; i ++)
+	{
+		#ifdef DEBUG
+		printf("%i\n", i);
+		#endif
+		t_mapss *map = ft_mapss_gen(tests[i].content, tests[i].sz);
+		assert(map);
+		catch("custom.tmp", &out, &save);
+		ft_mapss_print(map);
```

Comment:
how can i see results of this printing function?
the output redirected to "custom.tmp". i commented `line 86 remove("custom.tmp")` but dont see created file in directory with executed file.

Reply:
```
cd libft && make fulltest
gdbtui ./tests/test
breakpoint ft_mapss_print_test.c:81
start
continue
```
then see in `libft` folder

Reply:
also you can try and call in manually from tania_main :) just copy some generation from here, but without output interception

Reply:
No, please do check :)

Reply:
ok, it works :)

```
Map size: [3]
key: [], value: [ccc]
key: [a], value: [aaa]
key: [b], value: []
```
but why don't i see file if i compile without remove("custo.tmp");